x-pack/plugins/ml/server/lib/alerts/jobs_health_service.ts

```diff
@@ -0,0 +1,175 @@
+import { ALL_JOBS_SELECTION } from '../../../common/constants/alerts';
+import { MlJobsResponse } from '../../../common/types/job_service';
+import { DatafeedStats } from '../../../common/types/anomaly_detection_jobs';
+import { GetGuards } from '../../shared_services/shared_services';
+import { AnomalyDetectionJobsHealthAlertContext } from './register_jobs_monitoring_rule_type';
+import { getResultJobsHealthRuleConfig } from '../../../common/util/alerts';
+import { JobsHealthTests } from '../../../common/types/alerts';
+
+interface TestResult {
+  name: JobsHealthTests;
+  context: AnomalyDetectionJobsHealthAlertContext;
+}
+
+type TestsResults = TestResult[];
+
+type NotStartedDatafeedResponse = Array<DatafeedStats & { job_id: string }>;
+
+export function jobsHealthServiceProvider(
+  mlClient: MlClient,
+  datafeedsService: DatafeedsService,
+  logger: Logger
+) {
+  /**
+   * Extracts result list of job ids based on included and excluded selection of jobs and groups.
+   * @param includeJobs
+   * @param excludeJobs
+   */
+  const getResultJobIds = async (includeJobs: JobSelection, excludeJobs?: JobSelection | null) => {
+    const jobAndGroupIds = [...(includeJobs.jobIds ?? []), ...(includeJobs.groupIds ?? [])];
+
+    const includeAllJobs = jobAndGroupIds.some((id) => id === ALL_JOBS_SELECTION);
+
+    // Extract jobs from group ids and make sure provided jobs assigned to a current space
+    const jobsResponse = (
+      await mlClient.getJobs<MlJobsResponse>({
+        ...(includeAllJobs ? {} : { job_id: jobAndGroupIds }),
+      })
+    ).body.jobs;
+
+    let resultJobIds = jobsResponse.map((v) => v.job_id);
+
+    if (excludeJobs && (!!excludeJobs.jobIds.length || !!excludeJobs?.groupIds.length)) {
+      const excludedJobAndGroupIds = [
+        ...(excludeJobs?.jobIds ?? []),
+        ...(excludeJobs?.groupIds ?? []),
+      ];
+      const excludedJobsResponse = (
+        await mlClient.getJobs<MlJobsResponse>({
+          job_id: excludedJobAndGroupIds,
+        })
+      ).body.jobs;
+
+      const excludedJobsIds: Set<string> = new Set(excludedJobsResponse.map((v) => v.job_id));
+
+      resultJobIds = resultJobIds.filter((v) => !excludedJobsIds.has(v));
+    }
+
+    return resultJobIds;
+  };
+
+  return {
+    /**
+     * Gets not started datafeeds for provided jobs selection.
+     * @param jobIds
+     */
+    async getNotStartedDatafeeds(jobIds: string[]): Promise<NotStartedDatafeedResponse | void> {
+      const datafeeds = await datafeedsService.getDatafeedByJobId(jobIds);
+
+      if (datafeeds) {
+        const {
+          body: { datafeeds: datafeedsStats },
+        } = await mlClient.getDatafeedStats({
+          datafeed_id: datafeeds.map((d) => d.datafeed_id).join(','),
+        });
+
+        // match datafeed stats with the job ids
+        return (datafeedsStats as DatafeedStats[])
+          .filter((datafeedStat) => datafeedStat.state !== 'started')
```

Comment:
Just to double check - do we also need to check for `job state !== OPENED`?

Reply:
added check for the job state in https://github.com/elastic/kibana/pull/106084/commits/599d122fa1bb25de1faa916067f9c7812cc1e9fa

Reply:
I don't think this is correct. Should the job state condition be an `OR` ?

```
datafeedStat.state !== 'started' || datafeedStat.job_state === 'opened'
```

Reply:
changed in a6e0f92c7a233ab690c8699232532e229b4d040a